base_layer/core/src/block.rs

```diff
@@ -29,18 +29,66 @@ use crate::{
 };
 
 /// A Tari block. Blocks are linked together into a blockchain.
-pub struct Block {
+pub struct Block<'a> {
     pub header: BlockHeader,
-    pub body: AggregateBody,
+    pub body: AggregateBody<'a>,
 }
 
 /// The components of the block or transaction. The same struct can be used for either, since in Mimblewimble,
 /// cut-through means that blocks and transactions have the same structure.
-pub struct AggregateBody {
+pub struct AggregateBody<'a> {
```

Comment:
Perhaps a better name for this is BlockBody

Reply:
but it is also a Transaction body in addition to be a Block body which is why I think Cayle went with the more generic term. So there is a subtle difference between a Block and Transaction body (only 1 kernel in a transaction) but other than that difference everything else between them is common so I feel that separating the two bodies into distinct structs will create bloat.